Can't build docker file for arm32v7

**Describe the bug**

I need a onnxruntime build for c++ on arm32v7.
I am follwing the build instructions from here https://onnxruntime.ai/docs/build/inferencing.html#arm
and here https://github.com/microsoft/onnxruntime/blob/master/dockerfiles/README.md#arm-32v7

I cloned the git repo  https://github.com/microsoft/onnxruntime.git  and executed "git submodule update --init".
When I execute the "docker build -t onnxruntime-source-arm32v7 -f Dockerfile.arm32v7 .." the docker build fails saying 

 build [ERROR] - Failed to resolve executable path for 'cmake'.

**Urgency**
none

**System information**
 - Windows 10 Pro 19043.1415
 - Docker version 20.10.8, build 3967b7d
 - i7 8th gen / 32 GB Ram

**To Reproduce**
Follow the docker build instructions as stated in the description

**Expected behavior**
The docker build should proceed without errors and create the libonnxruntime.so to be used on linux/arm32

**Screenshots**
![image](https://user-images.githubusercontent.com/10888825/149503347-4258a7ee-ba56-4f06-8fca-0efef4d8e0e0.png)


**Additional context**
When I do the same on Ubuntu 20.04 it is even worse since it can't execute "RUN dnf .... command" in script "install_fedora_arm32.sh"
It is saying "standard_init_linux.go:228 exec user process caused: exec format error"

--- a/dockerfiles/Dockerfile.arm32v7
+++ b/dockerfiles/Dockerfile.arm32v7
@@ -8,8 +8,8 @@ FROM arm32v7/fedora:34
 MAINTAINER Changming Sun "chasun@microsoft.com"
 ADD . /code
 
-
-RUN /code/dockerfiles/scripts/install_fedora_arm32.sh && cd /code && ./build.sh --skip_submodule_sync --config Release --build_wheel --update --build --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER) 
+RUN /code/dockerfiles/scripts/install_fedora_arm32.sh
+RUN cd /code && ./build.sh --skip_submodule_sync --config Release --build_wheel --update --build --parallel --cmake_extra_defines ONNXRUNTIME_VERSION=$(cat ./VERSION_NUMBER)
 
 FROM arm64v8/centos:7
 COPY --from=0 /code/build/Linux/Release/dist /root

--- a/dockerfiles/scripts/install_fedora_arm32.sh
+++ b/dockerfiles/scripts/install_fedora_arm32.sh
@@ -1,3 +1,5 @@
-dnf install -y binutils gcc gcc-c++ aria2 python3-pip python3-wheel git python3-devel
+#!/bin/bash
+set -e
+dnf install -y binutils gcc gcc-c++ aria2 python3-pip python3-wheel git python3-devel cmake
 python3 -m pip install --upgrade pip
 python3 -m pip install numpy

--- a/include/onnxruntime/core/common/eigen_common_wrapper.h
+++ b/include/onnxruntime/core/common/eigen_common_wrapper.h
@@ -18,6 +18,13 @@
 #ifdef HAS_DEPRECATED_COPY
 #pragma GCC diagnostic ignored "-Wdeprecated-copy"
 #endif
+// cmake/external/eigen/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/NEON/PacketMath.h:1633:9:
+// error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘Eigen::internal::Packet4c’
+// {aka ‘struct Eigen::internal::eigen_packet_wrapper<int, 2>’} from an array of ‘const int8_t’
+// {aka ‘const signed char’} [-Werror=class-memaccess]
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 #elif defined(_MSC_VER)
 // build\windows\debug\external\eigen3\unsupported\eigen\cxx11\src/Tensor/Tensor.h(76):
 // warning C4554: '&': check operator precedence for possible error; use parentheses to clarify precedence

--- a/include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h
+++ b/include/onnxruntime/core/platform/EigenNonBlockingThreadPool.h
@@ -20,6 +20,13 @@
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wunused-result"
+// cmake/external/eigen/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/NEON/PacketMath.h:1633:9:
+// error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘Eigen::internal::Packet4c’
+// {aka ‘struct Eigen::internal::eigen_packet_wrapper<int, 2>’} from an array of ‘const int8_t’
+// {aka ‘const signed char’} [-Werror=class-memaccess]
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 #elif defined(_MSC_VER)
 #pragma warning(push)
 #pragma warning(disable : 4127)

--- a/onnxruntime/core/util/math_cpuonly.h
+++ b/onnxruntime/core/util/math_cpuonly.h
@@ -35,6 +35,13 @@
 #pragma GCC diagnostic ignored "-Wdeprecated-copy"
 #endif
 #endif
+// cmake/external/eigen/Eigen/src/Core/arch/NEON/PacketMath.h:1633:9:
+// error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘Eigen::internal::Packet4c’
+// {aka ‘struct Eigen::internal::eigen_packet_wrapper<int, 2>’} from an array of ‘const int8_t’
+// {aka ‘const signed char’} [-Werror=class-memaccess]
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 #elif defined(_MSC_VER)
 // build\windows\debug\external\eigen3\unsupported\eigen\cxx11\src/Tensor/Tensor.h(76):
 // warning C4554: '&': check operator precedence for possible error; use parentheses to clarify precedence

--- a/onnxruntime/test/perftest/performance_runner.cc
+++ b/onnxruntime/test/perftest/performance_runner.cc
@@ -20,10 +20,18 @@ using onnxruntime::Status;
 
 // TODO: Temporary, while we bring up the threadpool impl...
 #include "core/platform/threadpool.h"
+#include "onnxruntime_config.h"
 #if defined(__GNUC__)
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wunused-result"
+// cmake/external/eigen/unsupported/Eigen/CXX11/../../../Eigen/src/Core/arch/NEON/PacketMath.h:1633:9:
+// error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘Eigen::internal::Packet4c’
+// {aka ‘struct Eigen::internal::eigen_packet_wrapper<int, 2>’} from an array of ‘const int8_t’
+// {aka ‘const signed char’} [-Werror=class-memaccess]
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 #endif
 #include <unsupported/Eigen/CXX11/ThreadPool>
 #if defined(__GNUC__)

--- a/onnxruntime/test/util/compare_ortvalue.cc
+++ b/onnxruntime/test/util/compare_ortvalue.cc
@@ -10,6 +10,13 @@
 #pragma GCC diagnostic ignored "-Wignored-qualifiers"
 #pragma GCC diagnostic ignored "-Wunused-parameter"
 #pragma GCC diagnostic ignored "-Wunused-result"
+// cmake/external/eigen/Eigen/src/Core/arch/NEON/PacketMath.h:1633:9:
+// error: ‘void* memcpy(void*, const void*, size_t)’ copying an object of non-trivial type ‘Eigen::internal::Packet4c’
+// {aka ‘struct Eigen::internal::eigen_packet_wrapper<int, 2>’} from an array of ‘const int8_t’
+// {aka ‘const signed char’} [-Werror=class-memaccess]
+#ifdef HAS_CLASS_MEMACCESS
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+#endif
 #endif
 #include <google/protobuf/message_lite.h>
 #include <Eigen/Core>

